tests/assembly/targets.rs

```diff
@@ -0,0 +1,700 @@
+// [x86_64_unknown_fuchsia]: needs-llvm-components: x86
+// revisions: x86_64_unknown_haiku
+// [x86_64_unknown_haiku]: compile-flags: --target x86_64-unknown-haiku
+// [x86_64_unknown_haiku]: needs-llvm-components: x86
+// revisions: x86_64_unknown_hermit
+// [x86_64_unknown_hermit]: compile-flags: --target x86_64-unknown-hermit
+// [x86_64_unknown_hermit]: needs-llvm-components: x86
+// revisions: x86_64_unknown_illumos
+// [x86_64_unknown_illumos]: compile-flags: --target x86_64-unknown-illumos
+// [x86_64_unknown_illumos]: needs-llvm-components: x86
+// revisions: x86_64_unknown_l4re_uclibc
+// [x86_64_unknown_l4re_uclibc]: compile-flags: --target x86_64-unknown-l4re-uclibc
+// [x86_64_unknown_l4re_uclibc]: needs-llvm-components: x86
+// revisions: x86_64_unknown_linux_gnu
+// [x86_64_unknown_linux_gnu]: compile-flags: --target x86_64-unknown-linux-gnu
+// [x86_64_unknown_linux_gnu]: needs-llvm-components: x86
+// revisions: x86_64_unknown_linux_gnux32
+// [x86_64_unknown_linux_gnux32]: compile-flags: --target x86_64-unknown-linux-gnux32
+// [x86_64_unknown_linux_gnux32]: needs-llvm-components: x86
+// revisions: x86_64_unknown_linux_musl
+// [x86_64_unknown_linux_musl]: compile-flags: --target x86_64-unknown-linux-musl
+// [x86_64_unknown_linux_musl]: needs-llvm-components: x86
+// revisions: x86_64_unknown_linux_ohos
+// [x86_64_unknown_linux_ohos]: compile-flags: --target x86_64-unknown-linux-ohos
+// [x86_64_unknown_linux_ohos]: needs-llvm-components: x86
+// revisions: x86_64_unknown_netbsd
+// [x86_64_unknown_netbsd]: compile-flags: --target x86_64-unknown-netbsd
+// [x86_64_unknown_netbsd]: needs-llvm-components: x86
+// revisions: x86_64_unknown_none
+// [x86_64_unknown_none]: compile-flags: --target x86_64-unknown-none
+// [x86_64_unknown_none]: needs-llvm-components: x86
+// revisions: x86_64_unknown_openbsd
+// [x86_64_unknown_openbsd]: compile-flags: --target x86_64-unknown-openbsd
+// [x86_64_unknown_openbsd]: needs-llvm-components: x86
+// revisions: x86_64_unknown_redox
+// [x86_64_unknown_redox]: compile-flags: --target x86_64-unknown-redox
+// [x86_64_unknown_redox]: needs-llvm-components: x86
+// revisions: x86_64_unknown_uefi
+// [x86_64_unknown_uefi]: compile-flags: --target x86_64-unknown-uefi
+// [x86_64_unknown_uefi]: needs-llvm-components: x86
+// revisions: x86_64_uwp_windows_gnu
+// [x86_64_uwp_windows_gnu]: compile-flags: --target x86_64-uwp-windows-gnu
+// [x86_64_uwp_windows_gnu]: needs-llvm-components: x86
+// revisions: x86_64_uwp_windows_msvc
+// [x86_64_uwp_windows_msvc]: compile-flags: --target x86_64-uwp-windows-msvc
+// [x86_64_uwp_windows_msvc]: needs-llvm-components: x86
+// revisions: x86_64_wrs_vxworks
+// [x86_64_wrs_xworks]: compile-flags: --target x86_64-wrs-xworks
+// [x86_64_wrs_xworks]: needs-llvm-components: x86
+// revisions: x86_64h_apple_darwin
+// [x86_64h_apple_darwin]: compile-flags: --target x86_64h-apple-darwin
+// [x86_64h_apple_darwin]: needs-llvm-components: x86
```

Comment:
...Was this auto-generated? It looks like tidy is perhaps enforcing all of these lines exist, maybe we can add a --bless-ability to that?

I guess when targets are added it's fairly easy to update this list, and we remove targets relatively rarely.

Reply:
It was vaguely auto-generated, a handful of editor macros on the output from `ls`. I don't know that we need bless-ability for this quite yet, tidy will enforce that targets are added when they're added to the compiler, and the test will fail when targets are removed because the targets are removed. Both happen infrequently enough that it should be fine.